ddtrace/tracer/tracer.go

```diff
@@ -45,6 +47,13 @@ type tracer struct {
 	// stopped is a channel that will be closed when the worker has exited.
 	stopped chan struct{}
 
+	// stopping synchronizes calls to Stop() to ensure the tracer is stopped
```

Comment:
I think it'd be nice to not need this, considering all the other synchronization going on.
However, that can be refactored out in a future PR.

Reply:
👍 For now we'll keep it. It only protects against races between multiple goroutines calling `Stop` at the same time.